src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,157 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl service;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+        user = new User();
+    }
+
+    @Test
+    void register_newUser_Ok() {
+        user.setAge(19);
+        user.setLogin("JohnCoolBoyyyy");
+        user.setPassword("val1dPassw0rd11");
+        assertEquals(user, service.register(user));
+    }
+
+    @Test
+    void register_userExist_notOk() {
+        user.setLogin("validUserName");
+        assertThrows(InvalidDataException.class, () -> service.register(user));
```

Comment:
In the `register_userExist_notOk` test, you should first register a user with the same login to ensure the user already exists in the storage before calling `service.register(user)`. This will make the test case more robust and accurately test the scenario where a user already exists.